$OpenBSD: patch-source_common_putilimp_h,v 1.11 2017/06/07 09:58:30 ajacoutot Exp $

hppa doesn't have atomic builtins:
__sync_val_compare_and_swap, __sync_add_and_fetch, __sync_sub_and_fetch
don't assume having GNUC >= 404 implies they're here...

Index: source/common/putilimp.h
--- source/common/putilimp.h.orig
+++ source/common/putilimp.h
@@ -193,7 +193,7 @@ typedef size_t uintptr_t;
     /* Use the predefined value. */
 #elif U_PLATFORM == U_PF_MINGW
     #define U_HAVE_GCC_ATOMICS 0
-#elif U_GCC_MAJOR_MINOR >= 404 || defined(__clang__)
+#elif U_GCC_MAJOR_MINOR >= 404 && !defined(__hppa__) || defined(__clang__)
     /* TODO: Intel icc and IBM xlc on AIX also support gcc atomics.  (Intel originated them.)
      *       Add them for these compilers.
      * Note: Clang sets __GNUC__ defines for version 4.2, so misses the 4.4 test here.
